### DISCUSSION TITLE
How to control SUMO vehicle in Webots

### DISCUSSION THREAD

**Discussion by zhangcong0315** [cyberbotics/webots#4584] [General]
How to control sumo vehicle in Webots? I tried to transform sumo vehicle from "solid" to "robot" and use controller to control it. But it does not work and report the keyerror: robot. How can we use Traci to control sumo vehicle in the Webots environment?
**Expected behavior**

**Screenshots**

**System**
 - Operating System: [e.g., Windows 10, Linux Ubuntu 18.04, macOS Mojave]
 - Graphics Card: [e.g., NVIDIA GeForce RTX 2080 11 GB, AMD Radeon RX 580 8GB, etc.]

**Additional context**

**Comment by stefaniapedrazzi** [comment]
To control sumo vehicles in Webots using traci you need a Supervisor controller that updates the position and orientation of the vehicles.
So the vehicle can be simple `Solid` nodes and you don't need to transform them to `Robot` or to implement a controller program for each vehicle.

I would suggest you to check these sample simulation distributed within the Webots library:
`projects/vehicles/worlds/sumo_interface_example.wbt`
`project/vehicles/projects/worlds/highway_overtake.wbt`
